tests/Api/ApiTestCase.php

```diff
@@ -75,53 +75,83 @@ protected function getConfiguration()
 
     /**
      * Assert that all the expected data of a content of a resource are the same in an actual one.
-     * The actual content can contains more data than the expected one.
+     * An associative array can contain more elements than expected, but an numeric key array must be strictly identical.
      *
      * @param array $expectedContent
      * @param array $actualContent
      */
-    public function assertSameContent(array $expectedContent, array $actualContent)
+    protected function assertSameContent(array $expectedContent, array $actualContent)
     {
-        $differences = $this->computeArrayDiff($expectedContent, $actualContent);
-
-        if (! empty($differences)) {
-            $this->fail(
-                'Failed asserting that the content has the expected body.'
-                .PHP_EOL
-                .'Differences between expected and actual content :'
-                .PHP_EOL
-                .var_export($differences, true)
-            );
-        }
+        $expectedContent = $this->sortResourceContent($expectedContent);
+        $actualContent = $this->sortResourceContent($actualContent);
+
+        $expectedContent = $this->mergeResourceContents($actualContent, $expectedContent);
+
+        $this->assertSame($expectedContent, $actualContent);
     }
 
     /**
-     * @param array $expectedValues
-     * @param array $actualValues
+     * Recursively merge an expected content in a actual one to be able to compare them.
+     * Numeric key arrays are kept identical.
+     *
+     * @param array $actualContent
+     * @param array $expectedContent
      *
      * @return array
      */
-    protected function computeArrayDiff(array $expectedValues, array $actualValues)
+    private function mergeResourceContents(array $actualContent, array $expectedContent)
+    {
+        foreach ($expectedContent as $key => $value) {
+            if (is_array($value) && isset($actualContent[$key]) && is_array($actualContent[$key])) {
+                $expectedContent[$key] = $this->mergeResourceContents($actualContent[$key], $expectedContent[$key]);
+            }
+        }
+
+        if ($this->isAssociativeArray($expectedContent)) {
+            $mergedContent = array_merge($actualContent, $expectedContent);
+        } else {
+            $mergedContent = $expectedContent;
```

Comment:
Nope. For a numeric keys array we have to keep the expected one intact. The merged content is the one who will compared with the actual.